public/app/features/dashboard/components/ShareModal/SharePublicDashboard.tsx

```diff
@@ -1,60 +1,64 @@
 import React, { useState, useEffect } from 'react';
 
-import { Button, Field, Switch } from '@grafana/ui';
+import { Button, Field, Switch, Alert } from '@grafana/ui';
 import { notifyApp } from 'app/core/actions';
 import { createErrorNotification, createSuccessNotification } from 'app/core/copy/appNotification';
+import { contextSrv } from 'app/core/services/context_srv';
+import { DashboardModel } from 'app/features/dashboard/state/DashboardModel';
 import { dispatch } from 'app/store/store';
 
 import {
   dashboardCanBePublic,
   getPublicDashboardConfig,
   savePublicDashboardConfig,
+  getDashboard,
   PublicDashboardConfig,
 } from './SharePublicDashboardUtils';
 import { ShareModalTabProps } from './types';
 
 interface Props extends ShareModalTabProps {}
 
-// 1. write test for dashboardCanBePublic
-// 2. figure out how to disable the switch
-
 export const SharePublicDashboard = (props: Props) => {
-  const [publicDashboardConfig, setPublicDashboardConfig] = useState<PublicDashboardConfig>({ isPublic: false });
   const dashboardUid = props.dashboard.uid;
+  const orgId = contextSrv.user.orgId;
+  const [publicDashboardConfig, setPublicDashboardConfig] = useState<PublicDashboardConfig>({
+    isPublic: false,
+    publicDashboard: { uid: '', dashboardUid, orgId },
+  });
+
+  const [dashboard, setDashboard] = useState<DashboardModel>({} as DashboardModel);
 
   useEffect(() => {
-    getPublicDashboardConfig(dashboardUid)
-      .then((pdc: PublicDashboardConfig) => {
-        setPublicDashboardConfig(pdc);
-      })
-      .catch(() => {
-        dispatch(notifyApp(createErrorNotification('Failed to retrieve public dashboard config')));
-      });
+    // dashboard model may be stale, get a fresh copy
+    getDashboard(dashboardUid, setDashboard).catch();
```

Comment:
the dashbord model should not be stale, and Grafana live will warn if two are editing same dashboard, and can do live updates/refresh of the dashboard if someone else make a change. The version column also prevents overwrites.

Reply:
since you fetching this dashboard from the backend it will actually be stale, the frontend dashboard state could have template variables added but not yet saved, so loading it from the backend here is actually causing the inverse of what you are trying to accomplish.  

To get the current variable list call props.dashboard.getVariables()  that will fetch the latest variable state from the currently opened dashboard.

Reply:
fixed